src/app/controllers/produto-controller.js

```diff
@@ -1,65 +1,140 @@
-var Produto = require("../models/produto");
+const produto = require("../models/produto");
 
 class ProdutoController {
   lista() {
-    return (req, resp) => {
-      Produto.find((err, produtos) => {
-        if (err)
-          return resp.status(400).send(`Erro ao listar os produtos. ${err}`);
-        resp.status(200).json(produtos);
-      });
+    return async (req, resp) => {
+      try {
+        const prod = await produto.findAll({
+          attributes: ["id", "nome", "valor", "descricao"]
+        });
+        resp.status(200).json(prod);
+      } catch (error) {
+        return resp
+          .status(400)
+          .json({ message: `Erro ao listar os produtos. ${error}` });
+      }
+
+      // Produto.find((err, produtos) => {
+      //   if (err)
+      //     return resp.status(400).send(`Erro ao listar os produtos. ${err}`);
+      //   resp.status(200).json(produtos);
+      // });
     };
   }
 
   cadastra() {
-    return (req, resp) => {
-      var produto = new Produto();
-      //TODO: Assert fields
-      produto.nome = req.body.nome;
-      produto.preco = req.body.preco;
-      produto.descricao = req.body.descricao;
-
-      produto.save(err => {
-        if (err) return resp.status(400).send(`Erro ao salvar produto: ${err}`);
+    return async (req, resp) => {
+      try {
+        await produto.create({
+          nome: req.body.nome,
+          valor: req.body.valor,
+          descricao: req.body.descricao
+        });
         resp.status(201).json({ message: "Produto cadastrado com sucesso!!!" });
-      });
+      } catch (error) {
+        return resp
+          .status(400)
+          .json({ message: `Erro ao cadastrar o produto. ${error}` });
```

Comment:
Se esse `error` for um objeto ele mostra ou quebra? Não  lembro disso

Reply:
Mostra... não esta quebrando.

Reply:
Show